### PR TITLE
GNNE-1781 runtime kernel tests

### PR DESCRIPTION
**Pull request by Hejunchao6**

### CODE REVIEW

**Comment by xhuohai** [comment]
Please add Jira ID to the title.

**Comment by Hejunchao6** [comment]
> Please add Jira ID to the title.

done

**Review by liuzm6217-jianan — APPROVED** [review]

File: tests/kernels/test_batch_normalization.cpp [hunk 1]
```diff
@@ -101,7 +101,7 @@ TEST_P(BatchNormalizationTest, batch_normalization) {
     void *ptr_ort = tensor_buffer(output_ort, &size);
     dims_t shape(tensor_rank(output_ort));
     tensor_shape(output_ort, reinterpret_cast<int64_t *>(shape.data()));
-    auto expected = hrt::create(nncase::dt_float32, shape,
+    auto expected = hrt::create(input.datatype(), shape,
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_batch_normalization.cpp:104]: done

File: tests/kernels/test_topK.cpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                                     dims_t{1, 3}),
+                     testing::Values(0, -1), testing::Values(0, 1),
+                     testing::Values(0, 1)));
+
+TEST_P(TopKTest, TopK) {
+    auto l_ort = runtime_tensor_2_ort_tensor(input);
+
+    // expected
+    size_t size = 0;
+    int64_t k_array[] = {1};
```
Comment (xhuohai) [inline tests/kernels/test_topK.cpp:93]: k也要作为测试的参数

File: tests/kernels/test_unsqueeze.cpp [hunk 3]
```diff
@@ -73,12 +76,21 @@ TEST_P(UnsqueezeTest, Unsqueeze) {
 
     // actual
     auto output = kernels::stackvm::unsqueeze(input.impl(), axes.impl())
-                      .expect("squeeze failed");
+                      .expect("unsqueeze failed");
```
Comment (xhuohai) [inline tests/kernels/test_unsqueeze.cpp:79]: axes_array作为测试的参数

File: tests/kernels/kernel_test.h [hunk 4]
```diff
@@ -1163,8 +1517,10 @@ class KernelTest {
                     break;
                 }
                 case dt_boolean: {
-                    vec1.push_back(get<bool>(lhs, index) ? 2 : 1);
-                    vec2.push_back(get<bool>(rhs, index) ? 2 : 1);
+                    vec1.push_back(
+                        static_cast<float>(get<bool>(lhs, index) ? 2 : 1));
```
Comment (xhuohai) [inline tests/kernels/kernel_test.h:1521]: bool应该直接比较是否相等，cosine不太合适

File: tests/kernels/functions.sh [hunk 5]
```diff
@@ -0,0 +1,222 @@
+function GenerateHeader() {
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/functions.sh:1]: done

File: tests/kernels/kernel_test.h (hunk 4 above)
Comment (Hejunchao6) [reply tests/kernels/kernel_test.h:1521]: done

File: tests/kernels/test_instance_normalization.cpp [hunk 6]
```diff
@@ -70,7 +69,7 @@ TEST_P(InstanceNormalizationTest, instance_normalization) {
 
     // expected
     auto output_ort =
-        ortki_InstanceNormalization(l_ort, scale_ort, b_ort, 0.01f);
+        ortki_InstanceNormalization(l_ort, scale_ort, b_ort, 1e-4f);
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_instance_normalization.cpp:72]: done

File: tests/kernels/test_lrn.cpp [hunk 7]
```diff
@@ -47,14 +47,14 @@ class LrnTest
...
-                                                                 16})));
+                                          testing::Values(dims_t{1, 3, 16, 16},
+                                                          dims_t{1, 3, 8, 8})));
 
 TEST_P(LrnTest, lrn) {
     auto l_ort = runtime_tensor_2_ort_tensor(input);
 
     // expected
-    auto output_ort = ortki_LRN(l_ort, 0.001f, 0.5f, 0.8f, 3);
+    auto output_ort = ortki_LRN(l_ort, 0.22f, 0.20f, 0.75f, 3);
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_lrn.cpp:57]: done

File: tests/kernels/test_space_to_batch.cpp [hunk 8]
```diff
@@ -26,56 +26,34 @@ using namespace nncase;
...
-        lhs = hrt::create(typecode, l_shape, host_runtime_tensor::pool_cpu_only)
-                  .expect("create tensor failed");
-        init_tensor(lhs);
-
-        rhs = hrt::create(typecode, r_shape, host_runtime_tensor::pool_cpu_only)
-                  .expect("create tensor failed");
-        init_tensor(rhs);
+        // expected
+        float_t expected_array[] = {1, 3, 9,  11, 2, 4, 10, 12,
+                                    5, 7, 13, 15, 6, 8, 14, 16};
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_space_to_batch.cpp:38]: onnx上没这个算子

File: tests/kernels/test_dequantize.cpp [hunk 9]
```diff
@@ -45,22 +45,31 @@ class DequantizeTest
     runtime_tensor input;
 };
 
-INSTANTIATE_TEST_SUITE_P(Dequantize, DequantizeTest,
-                         testing::Combine(testing::Values(dt_int8),
+INSTANTIATE_TEST_SUITE_P(dequantize, DequantizeTest,
+                         testing::Combine(testing::Values(dt_uint8, dt_int8),
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_dequantize.cpp:49]: 不支持,有bug,已提出

File: tests/kernels/test_quantize.cpp [hunk 10]
```diff
@@ -54,45 +56,60 @@ TEST_P(QuantizeTest, quantize) {
...
     tensor_shape(output_ort, reinterpret_cast<int64_t *>(shape.data()));
-    auto expected = hrt::create(input.datatype(), shape,
+    auto expected = hrt::create(dt_uint8, shape,
                                 {reinterpret_cast<gsl::byte *>(ptr_ort), size},
                                 true, host_runtime_tensor::pool_cpu_only)
                         .expect("create tensor failed");
 
     // actual
-    float_t quant_param[] = {127, 0.01f};
+    quant_param_t quantParam;
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_quantize.cpp:86]: 1.int8和int16都不支持的,已提出
2.done

File: tests/kernels/test_leaky_relu.cpp [hunk 11]
```diff
@@ -76,9 +77,18 @@ TEST_P(LeakyReluTest, leaky_relu) {
                       .expect("leaky_relu failed");
     runtime_tensor actual(output.as<tensor>().expect("as tensor failed"));
 
+    bool result = is_same_tensor(expected, actual) ||
+                  cosine_similarity_tensor(expected, actual);
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_leaky_relu.cpp:81]: done

File: tests/kernels/test_matmul.cpp [hunk 12]
```diff
@@ -50,11 +50,10 @@ class MatMulTest : public KernelTest,
 };
 
 INSTANTIATE_TEST_SUITE_P(
-    MatMul, MatMulTest,
+    mat_mul, MatMulTest,
     testing::Combine(
-        testing::Values(
-            dt_int32, dt_int64
-            /*, dt_float32*/), // todo dt_float32 case have some issue
+        testing::Values(dt_int32, dt_int64, dt_float32, dt_float64, dt_int32,
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_matmul.cpp:55]: done

File: tests/kernels/test_pad.cpp [hunk 13]
```diff
@@ -37,34 +37,36 @@ class PadTest
...
+        testing::Values(dt_int32, dt_int8, dt_uint8, dt_float64, dt_uint64,
+                        dt_int64, dt_float32, dt_float16, dt_uint32),
+        testing::Values(dims_t{1, 3, 24, 24}, dims_t{1, 3, 16, 16})));
 
 TEST_P(PadTest, Pad) {
 
     // expected
     size_t size = 0;
-    int64_t pad_ptr[] = {0, 0, 0, 1};
+    int64_t pad_ptr[] = {1, 0, 0, 0, 0, 0, 0, 0};
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_pad.cpp:63]: pytest中的用例中pad实际只支持这个
pad mode只支持constant

File: tests/kernels/test_split.cpp [hunk 14]
```diff
@@ -61,33 +72,40 @@ TEST_P(SplitTest, Split) {
...
-        ortki_Split(l_ort, runtime_tensor_2_ort_tensor(sextions), -3), 0);
-    void *ptr_ort = tensor_buffer(output_ort, &size);
-    dims_t shape(tensor_rank(output_ort));
-    tensor_shape(output_ort, reinterpret_cast<int64_t *>(shape.data()));
-    auto expected = hrt::create(input.datatype(), shape,
-                                {reinterpret_cast<gsl::byte *>(ptr_ort), size},
-                                true, host_runtime_tensor::pool_cpu_only)
-                        .expect("create tensor failed");
+    auto output_ort1 = tensor_seq_get_value(
+        ortki_Split(l_ort, runtime_tensor_2_ort_tensor(sextions), axis_value),
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_split.cpp:76]: done

File: tests/kernels/test_prelu.cpp [hunk 15]
```diff
@@ -78,9 +78,18 @@ TEST_P(PreluTest, Prelu) {
                       .expect("prelu failed");
     runtime_tensor actual(output.as<tensor>().expect("as tensor failed"));
 
+    bool result = is_same_tensor(expected, actual) ||
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_prelu.cpp:81]: 1.done
2.slope不支持array,只支持scalar

File: tests/kernels/test_expand.cpp [hunk 16]
```diff
@@ -45,11 +45,14 @@ class ExpandTest
...
-                                                          dt_int64, dt_uint8,
-                                                          dt_int8, dt_int16),
-                                          testing::Values(dims_t{3, 1})));
+INSTANTIATE_TEST_SUITE_P(
+    Expand, ExpandTest,
+    testing::Combine(testing::Values(dt_float32, dt_int32, dt_int64, dt_uint8,
+                                     dt_int8, dt_int16, dt_uint16, dt_uint32,
+                                     dt_uint64, dt_float16, dt_float64,
+                                     dt_boolean),
+                     testing::Values(dims_t{3, 1}, dims_t{1, 1},
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_expand.cpp:54]: done

File: tests/kernels/test_unsqueeze.cpp (hunk 3 above)
Comment (Hejunchao6) [reply tests/kernels/test_unsqueeze.cpp:79]: done

File: tests/kernels/test_broadcast.cpp [hunk 17]
```diff
@@ -71,8 +71,18 @@ TEST_P(BroadCastTest, BroadCast) {
                       .expect("broadcast failed");
     runtime_tensor actual(output.as<tensor>().expect("as tensor failed"));
 
+    bool result = is_same_tensor(expected, actual) ||
+                  cosine_similarity_tensor(expected, actual);
+
+    if (!result) {
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_broadcast.cpp:77]: done

File: tests/kernels/test_clamp.cpp [hunk 18]
```diff
@@ -46,24 +46,26 @@ class ClampTest
...
                                      dims_t{1, 3}, dims_t{8, 8},
-                                     dims_t{1, 3, 8})));
+                                     dims_t{1, 3, 8}, dims_t{16, 16}, dims_t{},
+                                     dims_t{16})));
 
 TEST_P(ClampTest, clamp) {
 
     // expected
-    float_t min1[] = {-7.0f};
+    float_t min1[] = {-1.0f};
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_clamp.cpp:61]: done

File: tests/kernels/test_topK.cpp [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+    runtime_tensor sorted;
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    TopK, TopKTest,
+    testing::Combine(testing::Values(dt_float32),
+                     testing::Values(dims_t{1, 2, 4, 8}, dims_t{1, 3, 16, 16},
+                                     dims_t{3, 3, 6}, dims_t{16, 16}, dims_t{1},
+                                     dims_t{1, 3}),
+                     testing::Values(0, -1), testing::Values(0, 1),
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_topK.cpp:85]: axes只支持0和-1

File: tests/kernels/test_normal.cpp [hunk 20]
```diff
@@ -28,66 +28,64 @@ using namespace ortki;
...
                                                                  16})));
 
 TEST_P(NormalTest, normal) {
 
     // expected
-    int64_t shape_ptr[] = {1, 3, 16, 16};
+    std::vector<int64_t> vec(shape_array.begin(), shape_array.end());
+    int64_t shape_ptr[4];
+    std::copy(vec.begin(), vec.end(), shape_ptr);
     auto output_ort = ortki_RandomNormal(1, 0.5f, 1.0f, 1.0f, shape_ptr, 4);
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_normal.cpp:78]: done

File: tests/kernels/test_normal_like.cpp [hunk 21]
```diff
@@ -45,10 +45,12 @@ class NormalLikeTest
     runtime_tensor input;
 };
 
-INSTANTIATE_TEST_SUITE_P(NormalLike, NormalLikeTest,
-                         testing::Combine(testing::Values(dt_float32),
-                                          testing::Values(dims_t{1, 3, 16,
-                                                                 16})));
+INSTANTIATE_TEST_SUITE_P(
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_normal_like.cpp:48]: done

File: tests/kernels/test_uniform.cpp [hunk 22]
```diff
@@ -28,66 +28,68 @@ using namespace ortki;
 
 class UniformTest
     : public KernelTest,
-      public ::testing::TestWithParam<std::tuple<nncase::typecode_t, dims_t>> {
+      public ::testing::TestWithParam<std::tuple<nncase::typecode_t, axes_t>> {
   public:
     void SetUp() override {
         auto &&[typecode, l_shape] = GetParam();
+        float_t high_array[] = {1.0f};
+        high = hrt::create(dt_float32, {1},
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_uniform.cpp:36]: done

File: tests/kernels/test_uniform_like.cpp [hunk 23]
```diff
@@ -51,7 +51,9 @@ class UniformLikeTest : public KernelTest,
 
 INSTANTIATE_TEST_SUITE_P(
     UniformLike, UniformLikeTest,
-    testing::Combine(testing::Values(dt_float32),
+    testing::Combine(testing::Values(dt_float32, dt_int32/*, dt_int16, dt_float64,
+                                     dt_int8, dt_uint8, dt_uint16, dt_uint32,
+                                     dt_uint64, dt_int64, dt_float16*/),
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_uniform_like.cpp:56]: done

File: tests/kernels/test_transpose.cpp [hunk 24]
```diff
@@ -47,8 +47,10 @@ class TransposeTest
 
 INSTANTIATE_TEST_SUITE_P(
     Transpose, TransposeTest,
-    testing::Combine(testing::Values(dt_float32, dt_int32, dt_int16, dt_int8,
-                                     dt_uint8),
+    testing::Combine(testing::Values(dt_float32, dt_int32, dt_int16, dt_float64,
+                                     dt_int8, dt_uint8, dt_uint16, dt_uint32,
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_transpose.cpp:51]: done

File: tests/kernels/test_reduce_arg_max.cpp [hunk 25]
```diff
@@ -26,70 +26,72 @@ using namespace nncase;
...
-                                                          dt_int64),
-                                          testing::Values(dims_t{1, 3, 16, 16},
-                                                          /*dims_t { 3, 16, 16
-                                                          }, dims_t { 16, 16 },
-                                                          dims_t { 16 },*/
-                                                          dims_t{1})));
+INSTANTIATE_TEST_SUITE_P(
+    ReduceArgMax, ReduceArgMaxTest,
+    testing::Combine(testing::Values(dt_float32), testing::Values(dt_int64),
+                     testing::Values(dims_t{1, 3, 16, 16}),
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_reduce_arg_max.cpp:80]: done

File: tests/kernels/test_reduce_arg_min.cpp [hunk 26]
```diff
@@ -28,79 +28,70 @@ using namespace ortki;
...
-                                                          /*dims_t { 3, 16, 16
-                                                          }, dims_t { 16, 16 },
-                                                          dims_t { 16 },*/
-                                                          dims_t{1})));
+INSTANTIATE_TEST_SUITE_P(
+    ReduceArgMin, ReduceArgMinTest,
+    testing::Combine(testing::Values(dt_float32), testing::Values(dt_int64),
+                     testing::Values(dims_t{1, 3, 16, 16}),
+                     testing::Values(dims_t{1}), testing::Values(-1, 0),
+                     testing::Values(1, 0), testing::Values(1, 0)));
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_reduce_arg_min.cpp:82]: done

File: tests/kernels/test_concat.cpp [hunk 27]
```diff
@@ -50,19 +50,19 @@ class BinaryTest
...
+                                     dims_t{1, 3, 16}, dims_t{1})));
+
+TEST_P(ConcatTest, Concat) {
     auto l_ort = runtime_tensor_2_ort_tensor(lhs);
     auto r_ort = runtime_tensor_2_ort_tensor(rhs);
+    OrtKITensor *ls_ort[2] = {l_ort, r_ort};
 
     // expected
-    auto output_ort = ortki_Xor(l_ort, r_ort);
+    auto output_ort = ortki_Concat(ls_ort, 2, 0);
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_concat.cpp:65]: done

File: tests/kernels/test_constant_of_shape.cpp [hunk 28]
```diff
@@ -53,26 +53,12 @@ class ConstantOfShapeTest
     runtime_tensor expected;
 };
 
-INSTANTIATE_TEST_SUITE_P(ConstantOfShape, ConstantOfShapeTest,
+INSTANTIATE_TEST_SUITE_P(constant_of_shape, ConstantOfShapeTest,
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_constant_of_shape.cpp:56]: done

File: tests/kernels/test_topK.cpp (hunk 2 above)
Comment (Hejunchao6) [reply tests/kernels/test_topK.cpp:93]: done

File: tests/kernels/test_lstm.cpp [hunk 29]
```diff
@@ -113,15 +113,42 @@ TEST_P(LstmTest, lstm) {
...
+    tensor_shape(tensor_seq_get_value(output_ort, 2),
+                 reinterpret_cast<int64_t *>(shape3.data()));
+    auto expected3 =
+        hrt::create(dt_float32, shape3,
+                    {reinterpret_cast<gsl::byte *>(ptr_ort3), size}, true,
+                    host_runtime_tensor::pool_cpu_only)
+            .expect("create tensor failed");
+
+    runtime_tensor expected[] = {expected1, expected2, expected3};
+    typecode_t dtypes[] = {dt_float32, dt_float32, dt_float32};
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_lstm.cpp:151]: 1.输出size只支持3
2.done
3.done

File: tests/kernels/test_get_item.cpp [hunk 30]
```diff
@@ -75,8 +74,18 @@ TEST_P(GetItemTest, get_item) {
                       .expect("get_item failed");
     runtime_tensor actual(output.as<tensor>().expect("as tensor failed"));
 
+    bool result = is_same_tensor(expected, actual) ||
+                  cosine_similarity_tensor(expected, actual);
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_get_item.cpp:78]: done

File: tests/kernels/test_conv2d.cpp [hunk 31]
```diff
@@ -69,9 +70,10 @@ TEST_P(Conv2DTest, conv2d) {
     auto bais_ort = runtime_tensor_2_ort_tensor(bais);
 
     // expected
-    const char *auto_pad = "NOTSET";
+    const char auto_pad[7] = "NOTSET";
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_conv2d.cpp:73]: done

File: tests/kernels/test_conv2d_transpose.cpp [hunk 32]
```diff
@@ -146,9 +146,18 @@ TEST_P(Conv2DTransposeTest, conv2d_transpose) {
             .expect("conv2d_transpose failed");
     runtime_tensor actual(output.as<tensor>().expect("as tensor failed"));
 
+    bool result = is_same_tensor(expected, actual) ||
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_conv2d_transpose.cpp:149]: done

File: tests/kernels/test_reduce_window_2d.cpp [hunk 33]
```diff
@@ -122,9 +122,18 @@ TEST_P(ReduceWindow2DTest, ReduceWindow2D) {
                       .expect("reduce_window_max failed");
     runtime_tensor actual(output.as<tensor>().expect("as tensor failed"));
 
+    bool result = is_same_tensor(expected, actual) ||
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_reduce_window_2d.cpp:125]: done

File: tests/kernels/test_scatter_nd.cpp [hunk 34]
```diff
@@ -28,57 +28,57 @@ using namespace ortki;
...
+        init_tensor(updates);
     }
 
     void TearDown() override {}
 
   protected:
     runtime_tensor input;
+    runtime_tensor indices;
+    runtime_tensor updates;
 };
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_scatter_nd.cpp:63]: done

File: tests/kernels/test_slice.cpp [hunk 35]
```diff
@@ -52,9 +52,12 @@ class SliceTest
     runtime_tensor input;
 };
 
-INSTANTIATE_TEST_SUITE_P(Slice, SliceTest,
-                         testing::Combine(testing::Values(dt_int32),
-                                          testing::Values(dims_t{2, 3, 4, 5})));
+INSTANTIATE_TEST_SUITE_P(
+    Slice, SliceTest,
+    testing::Combine(testing::Values(dt_int32),
+                     testing::Values(dims_t{2, 3, 4, 5}, dims_t{1, 4, 5, 6},
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_slice.cpp:58]: done

File: tests/kernels/test_shape_of.cpp [hunk 36]
```diff
@@ -42,13 +42,15 @@ class ShapeOfTest
...
 };
 
-INSTANTIATE_TEST_SUITE_P(ShapeOf, ShapeOfTest,
-                         testing::Combine(testing::Values(dt_float32),
-                                          testing::Values(dims_t{1, 3, 16,
-                                                                 16})));
+INSTANTIATE_TEST_SUITE_P(
+    ShapeOf, ShapeOfTest,
+    testing::Combine(testing::Values(dt_float32, dt_int32, dt_int8, dt_uint8,
+                                     dt_float64, dt_int16, dt_uint32, dt_uint64,
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_shape_of.cpp:50]: done

File: tests/kernels/test_resize_image.cpp [hunk 37]
```diff
@@ -26,98 +26,112 @@ using namespace nncase;
...
-                runtime::stackvm::image_resize_mode_t::bilinear,
-                runtime::stackvm::image_resize_transformation_mode_t::asymmetric,
-                runtime::stackvm::image_resize_nearest_mode_t::floor, false,
-                lhs.impl(), roi.impl(), expected.impl(), cubic_coeff_a.impl(),
-                exclude_outside.impl(), extrapolation_value.impl())
-                .expect("resize_image failed");
-        runtime_tensor actual(output.as<tensor>().expect("as tensor failed"));
+
+    auto output =
+        kernels::stackvm::resize_image(
```
[the remark this answers is not in the document]
Comment (Hejunchao6) [reply tests/kernels/test_resize_image.cpp:95]: done

**Pull request merged**